operators/kas-fleetshard-bundle-template.yaml

```diff
@@ -0,0 +1,73 @@
+---
+apiVersion: template.openshift.io/v1
+kind: Template
+metadata:
+  name: kas-fleetshard
+objects:
+- apiVersion: operators.coreos.com/v1alpha1
+  kind: CatalogSource
+  metadata:
+    name: kas-fleetshard-catalog
+    namespace: ${NAMESPACE}
+    labels:
+      app: kas-fleetshard
+  spec:
+    sourceType: grpc
+    secrets:
+    - "rhoas-image-pull-secret"
```

Comment:
@machi1990 - I don't think kas-fleet-manager supports a pull secret for the `CatalogSource`, correct? I'm doing a test with the standalone cluster configuration and the private index image cannot be pulled from osd-addons.

Reply:
Hi, it does. This https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/main/templates/secrets-template.yml#L137 was added for that. This done here:
- see https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/75ee96dc51150d2ae51a4a8eecc8ae036b1e2135/internal/kafka/internal/workers/clusters_mgr.go#L998
- The pull secret name is https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/75ee96dc51150d2ae51a4a8eecc8ae036b1e2135/internal/kafka/internal/workers/clusters_mgr.go#L48

Reply:
Thank you, I may have it configured incorrectly. I'll check it again.

Reply:
I checked back in KFM's standalone provider and I think this is where it would be configured for Strimzi:
https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/75ee96dc51150d2ae51a4a8eecc8ae036b1e2135/internal/kafka/internal/clusters/standalone_provider.go#L111-L114

And for kas-fleetshard:
https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/75ee96dc51150d2ae51a4a8eecc8ae036b1e2135/internal/kafka/internal/clusters/standalone_provider.go#L193-L196

The `CatalogSource` object accepts a string list called `Secrets` for use in pulling the bundle index itself.

Reply:
Ah right. The secrets is only created but not used. At the moment we can hardocde/use the created secret from https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/75ee96dc51150d2ae51a4a8eecc8ae036b1e2135/internal/kafka/internal/workers/clusters_mgr.go#L998 
Is there any other configuration that needs to be configured?

Reply:
A nice to have feature would be the ability to set the `Config` on the `SubscriptionSpec`:
https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/75ee96dc51150d2ae51a4a8eecc8ae036b1e2135/internal/kafka/internal/clusters/standalone_provider.go#L145-L151

This would allow developers to override environment vars within the bundle and also lower/raise resource requests and limits on the deployments (e.g. to deploy to smaller node sizes for development/testing).

Reply:
> A nice to have feature would be the ability to set the Config on the SubscriptionSpec:
https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/75ee96dc51150d2ae51a4a8eecc8ae036b1e2135/internal/kafka/internal/clusters/standalone_provider.go#L145-L151

> This would allow developers to override environment vars within the bundle and also lower/raise resource requests and limits on the deployments (e.g. to deploy to smaller node sizes for development/testing).

Perfect. Do you have an example of how this is done at the moment? I'll open a PR for the pull secret.